packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -276,6 +290,27 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
   async signAndBroadcastTransaction(signTxInput: SignTxInput<ETHSignTx>): Promise<string> {
     try {
       const { txToSign, wallet } = signTxInput
+      const assets = await getGeneratedAssetData()
+      const feeAsset = assets[this.getFeeAssetId()]
+      // If there is a mismatch between the current wallet's EVM chain ID and the adapter's chainId?
+      // Switch the chain on wallet before building/sending the Tx
+      if (supportsEthSwitchChain(wallet)) {
+        const walletEvmChainId = await (wallet as ETHWallet).ethGetChainId?.()
+        const adapterEvmChainId = fromChainId(this.chainId).chainReference
+        if (!bnOrZero(walletEvmChainId).isEqualTo(Number(adapterEvmChainId))) {
+          await (wallet as ETHWallet).ethSwitchChain?.({
+            chainId: utils.hexValue(Number(adapterEvmChainId)),
+            chainName: this.getDisplayName(),
+            nativeCurrency: {
+              name: feeAsset.name,
+              symbol: feeAsset.symbol,
+              decimals: 18,
+            },
+            rpcUrls: [this.getRpcUrl()],
+            blockExplorerUrls: [feeAsset.explorer],
+          })
+        }
+      }
       const txHash = await (wallet as ETHWallet)?.ethSendTx?.(txToSign)
```

Comment:
You didn't write this, but it should be inside the `supportsEthSwitchChain` type guard, then we can remove the cast.

Reply:
I'm all for being a friendly neighbor and help tidying the house while around - 9d090f1fe1b24d90f2184744c2b09fb8859483eb

Reply:
Update - reverted beeca56ed985dcc53a3c09c3d08367e06c547a32 this will require tests to be updated to test `signAndBroadcastTransaction` with a MM wallet, vs. a native one currently